atomic_reactor/utils/rpm.py

```diff
@@ -35,7 +35,7 @@ def get_rpm_list(tags=None, separator=';'):
     return rpms
 
 
-def rpm_qf_args(tags=None, separator=';'):
+def rpm_qf_args(tags=None, separator=';', output_list=False):
```

Comment:
Looks adding this new parameter `output_list` adds a new technical debt, which makes the function return value in another different type. Can we just refactor this function to return value in only one type, that is a list object, and update the existing lines calling `rpm_qf_args`?

Reply:
so this has to return either

"-qa --qf '%{NAME};%{VERSION};%{RELEASE};%{ARCH};%{EPOCH};%{SIZE};%{SIGMD5};%{BUILDTIME};%{SIGPGP:pgpsig};%{SIGGPG:pgpsig}\\n'"

or

['-qa', '--qf', '%{NAME};%{VERSION};%{RELEASE};%{ARCH};%{EPOCH};%{SIZE};%{SIGMD5};%{BUILDTIME};%{SIGPGP:pgpsig};%{SIGGPG:pgpsig}\n']

if I return list and join it it will be:

'-qa --qf %{NAME};%{VERSION};%{RELEASE};%{ARCH};%{EPOCH};%{SIZE};%{SIGMD5};%{BUILDTIME};%{SIGPGP:pgpsig};%{SIGGPG:pgpsig}\n'

which is wrong as that format has to be quoted

if I return string and split it:
['-qa', '--qf', "'%{NAME};%{VERSION};%{RELEASE};%{ARCH};%{EPOCH};%{SIZE};%{SIGMD5};%{BUILDTIME};%{SIGPGP:pgpsig};%{SIGGPG:pgpsig}\\n'"]

which is wrong again because format will have additional ' at the start 

so I'll either leave it as it is or make another method

Reply:
I don't understand, how did you get extra `'` ?

Reply:
Bottom line is that it's very bad practice to return two **completely** different types from the same function.

Reply:
because split quotes it

Reply:
done